feat: trim icons and transition SVG path elements

hacdias opened the pull request:
Just trimming the icons and adding a new one 😄 

/cc @olizilla

hacdias commented:
Just pushed a commit to put a transition into SVG paths.

hacdias commented:
Don't merge this yet @olizilla  😄

hacdias commented:
@olizilla now using the trimmed icons. I think it's ready to merge 😄

alanshaw commented on ipfs.css:
```diff
@@ -369,7 +369,8 @@ See: https://github.com/ipfs-shipyard/ipfs-css
 
-.transition-all {
+.transition-all,
+.transition-all path {
```
Is this change related? What does it do?

hacdias replied:
It also applies the transition to the SVG paths

olizilla commented:
@hacdias we don't need to add `path` to all the things.

http://jsbin.com/celezal/5/edit?html,css,output

The fill and stroke styles follow the cascade as long as nothing overrides them. The svgs need another pass over them to remove the inline styles, and unneeded classes and ids.

Of note, being able to just set a single fill or stroke colour and have it sensibly change the colour of an svg is depends on the construction of the svg. We'll need to eyeball them all, which sounds like we need an svg preview page. I can take a look at that later if your deep in desktop code.

hacdias commented:
@olizilla thanks for noticing. I thought I had tried and it didn't work. Probably I got something wrong.

I just cleaned the SVGs by removing the classes and IDs that weren't needed and removed the `fill` attribute so now they're black by default unless someone sets their color (which is the intended use I think).

This is the first week of my uni semester so I'm still getting on a new routine 😄 and I'll be a bit less active this week. Although, I won't continue Desktop until this PR is finished.

I'm now trying to finish [js-ipfs#1230](https://github.com/ipfs/js-ipfs/pull/1230) and trying to finish some homework.

Do you want to do the Preview Page in a separate PR or still here?

olizilla commented:
Enjoy uni! The svg preview page can wait. I'll have a quick look at the svgs and get this merged.

hacdias commented:
Thanks @olizilla 😄

hacdias commented:
Any developments on this? 😃

hacdias commented:
By the way, I noticed there are some icons that would fit the square really easily but they are a lot smaller like the power one.

hacdias commented:
Ping @alanshaw @olizilla 😄

olizilla commented:
@hacdias looking good. There are a few icons missing... i guess some of them don't make sense as glyph or stroke versions, but I just wanted to check with you if you had more that you were processing or if they just don't exist yet:

![screenshot 2018-03-05 17 56 26](https://user-images.githubusercontent.com/58871/36991142-bfb9d210-209e-11e8-9f68-4738e69c32ea.png)

cc: @akrych

akrych commented:
Hello:) yes, they are correct. I didn't make glyph version for all

hacdias commented:
@olizilla cool! You should commit that page too 😄

hacdias commented:
Cool! Thanks @olizilla! Could you ping me when you publish a new version, please? :smile:

olizilla commented:
@hacdias ipfs-css@0.3.0 is a thing now!

hacdias commented:
Thank yooou!
